torch_xla/csrc/ops/ops.cpp

```diff
@@ -791,6 +792,47 @@ NodePtr LogicalAnd(const Value& input, const Value& other) {
       std::move(lower_fn));
 }
 
+NodePtr NanToNum(const Value& input, c10::optional<double> nan,
+                 c10::optional<double> posinf, c10::optional<double> neginf) {
+  auto lower_fn = [nan, posinf, neginf](const Node& node,
+                                        LoweringContext* loctx) -> XlaOpVector {
+    xla::XlaOp xla_input = loctx->GetOutputOp(node.operand(0));
+    const xla::Shape& input_shape = XlaHelpers::ShapeOfXlaOp(xla_input);
+    auto element_type = input_shape.element_type();
+
+    if (!xla::primitive_util::IsFloatingPointType(element_type)) {
+      return node.ReturnOp(xla_input, loctx);
+    }
+
+    xla::XlaOp nan_replacement =
+        nan.has_value()
+            ? XlaHelpers::ScalarValue(*nan, element_type, xla_input.builder())
+            : xla::Zero(xla_input.builder(), element_type);
+    xla::XlaOp pos_inf_replacement =
+        posinf.has_value()
+            ? XlaHelpers::ScalarValue(*posinf, element_type,
+                                      xla_input.builder())
+            : xla::MaxFiniteValue(xla_input.builder(), element_type);
+    xla::XlaOp neg_inf_replacement =
+        posinf.has_value()
+            ? XlaHelpers::ScalarValue(*neginf, element_type,
+                                      xla_input.builder())
+            : xla::MinFiniteValue(xla_input.builder(), element_type);
```

Comment:
ditto, let's resolve the optional in `aten_xla_type.cpp`

Reply:
Changed.